### PR TITLE
Wip/wasm

### PR DESCRIPTION
**Pull request by fabiosky**
*Issue #, if available:*
https://github.com/aws/dcv-color-primitives/issues/23
https://github.com/aws/dcv-color-primitives/issues/35

*Description of changes:*
* Add to README that dcv-color-primitives C bindings requires to link also ws2_32.lib and userenv.lib.
* WebAssembly support

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by nacho** [comment]
Looks good to me

File: Cargo.toml [hunk 1]
```diff
@@ -16,9 +16,14 @@ crate-type = ["staticlib", "rlib"]
...
 criterion = "0.3.0"
-rand = "0.7.3"
 itertools = "0.8.2"
 
 [[bench]]
 name = "benches"
 harness = false
+
+[target.'cfg(target_arch = "wasm32")'.dev-dependencies]
+wasm-bindgen-test = "0.3.13"
```
Comment (Urhengulas) [inline Cargo.toml:26]: Maybe specifying `0.3` is enough? 
```suggestion
wasm-bindgen-test = "0.3"
```

File: Cargo.toml [hunk 2]
```diff
@@ -16,9 +16,14 @@ crate-type = ["staticlib", "rlib"]
...
+
+[target.'cfg(target_arch = "wasm32")'.dependencies]
+wasm-bindgen = "0.2.63"
```
Comment (Urhengulas) [inline Cargo.toml:29]: Also here `0.2`:
```suggestion
wasm-bindgen = "0.2"
```

**Comment by Urhengulas** [comment]
And a small other idea would be to use [cfg-if](https://docs.rs/cfg-if/0.1.10/cfg_if/index.html) for the conditional compilation. Could make it bit cleaner.

**Comment by Urhengulas** [comment]
Hello @fabiosky :wave: 

When do you think this PR will land? I'd like to commit it to our main branch and would prefer a cargo release over a git-dependency.

Are there any blockers I could help with?

File: Cargo.toml (hunk 1 above)
Comment (fabiosky) [reply Cargo.toml:26]: Changed as suggested

File: Cargo.toml (hunk 2 above)
Comment (fabiosky) [reply Cargo.toml:29]: Changed as suggested

**Comment by fabiosky** [comment]
Hi @Urhengulas ,

I rebased the code above on master,
I applied the changes you suggested, except for the cfg-if which will come in a future release.

**Comment by fabiosky** [comment]
Here's the new release that comes with wasm32 support:
https://github.com/aws/dcv-color-primitives/releases/tag/v0.1.14

**Comment by fabiosky** [comment]
Published on crates too:
https://crates.io/crates/dcv-color-primitives

**Comment by fabiosky** [comment]
This was merged on master.
Closing.

**Comment by Urhengulas** [comment]
Thank you very much for your work @fabiosky!! :smile:

**Pull request closed without merging**